src/words/TokTok.jl

```diff
@@ -0,0 +1,172 @@
+const CLOSE_PUNCT =[
+        "\u0f3b", "\u0f3d", "\u169c", "\u2046", "\u207e", "\u208e", "\u232a", "\uff63",
+        "\u2769", "\u276b", "\u276d", "\u276f", "\u2771", "\u2773", "\u2775", "\u27c6",
+        "\u27e7", "\u27e9", "\u27eb", "\u27ed", "\u27ef", "\u2984", "\u2986", "\u2988",
+        "\u298a", "\u298c", "\u298e", "\u2990", "\u2992", "\u2994", "\u2996", "\u2998",
+        "\u29d9", "\u29db", "\u29fd", "\u2e23", "\u2e25", "\u2e27", "\u2e29", "\u3009",
+        "\u300b", "\u300d", "\u300f", "\u3011", "\u3015", "\u3017", "\u3019", "\u301b",
+        "\u301e", "\u301f", "\ufd3f", "\ufe18", "\ufe36", "\ufe38", "\ufe3a", "\ufe3c",
+        "\ufe3e", "\ufe40", "\ufe42", "\ufe44", "\ufe48", "\ufe5a", "\ufe5c", "\ufe5e",
+        "\uff09", "\uff3d", "\uff5d", "\uff60"]
+    
+# This is the \p{Close_Punctuation} from Perl's perluniprops             
+const CURRENCY_SYM = [
+        "\xa2", "\xa3", "\xa4", "\xa5", "\u058f", "\u060b", "\u09f2", "\u09f3", "\u09fb",
+        "\u0af1", "\u0bf9", "\u0e3f", "\u17db", "\u20a0", "\u20a1", "\u20a2", "\u20a3",
+        "\u20a4", "\u20a5", "\u20a6", "\u20a7", "\u20a8", "\u20a9", "\u20aa", "\u20ab",
+        "\u20ac", "\u20ad", "\u20ae", "\u20af", "\u20b0", "\u20b1", "\u20b2", "\u20b3",
+        "\u20b4", "\u20b5", "\u20b6", "\u20b7", "\u20b8", "\u20b9", "\u20ba", "\ua838",
+        "\ufdfc", "\ufe69", "\uff04", "\uffe0", "\uffe1", "\uffe5", "\uffe6"]
+
+# Use for tokenizing URL-unfriendly characters: [:/?#]
+const URL_FOE_3 = [":", "/", "+", ".", "\r", "\n", "\t", "\f", "\v"], "/"
+const URL_FOE_4 = [" /"], "/"
+
+# Left/Right strip, i.e. remove heading/trailing spaces.
+const LSTRIP = [" "], ""           
+const RSTRIP = ["\r", "\n", "\t", "\f", "\v"], "\n"  
+# Merge multiple spaces.
+const ONE_SPACE = ["  "], " "
+
+
+const rules_atoms = [
+        FUNKY_PUNCT_1,
+        FUNKY_PUNCT_2,
+        EN_EM_DASHES,
+        PROB_SINGLE_QUOTES,
+        OPEN_PUNCT,
+        CLOSE_PUNCT,
+        CURRENCY_SYM
+    ]
+
+const rules_replaces = [
+	NON_BREAKING,
+        AMPERCENT,
+        TAB,
+        PIPE,
+        STUPID_QUOTES_1,
+        STUPID_QUOTES_2,
+        URL_FOE_3,
+        URL_FOE_4,
+        ONE_SPACE
+    ]
+
+function toktok_tokenize(instring::AbstractString)
+    ts = TokenBuffer(instring)
+    isempty(input) && return ts.tokens
+ 
+    flag = length(ts.input)
+    # handles FINAL_PERIOD_1 = r"(?<!\.)\.$"
+    if ts.input[end-1 : end] != ".."
+        flush!(ts, ".")
+        flag -= 1
+    end
+
+    # handles FINAL_PERIOD_2 = r"(?<!\.)\.\s*(["'’»›”]) *$"
+    if ts.input[end] in String.split("“”‘’›")
+        i = flag - 1
+        while ts.input[i] in RSTRIP[1]
+            i -= 1
+        end
+        if ts.input[i] == "."
+            flag = i - 1
+            flush!(ts, ts.input[i:end])
+        end
+    end
+ 
+    while !isdone(ts) && ts.idx <= flag
+       atoms(ts, vcat(rules_atoms...)) || replaces(ts, vcat(rules_replaces...)) || replaces(ts, vcat(LSTRIP, RSTRIP), boundary = true)
```

Comment:
```suggestion
       atoms(ts, vcat(rules_atoms...)) ||
       replaces(ts, vcat(rules_replaces...)) || 
       replaces(ts, vcat(LSTRIP, RSTRIP), boundary = true)
```

Does `replaces` use a kwarg or a positional arg for  boundry?
It should match `lookahead`

Reply:
I'm not familiar with that :sweat_smile:  , could u review the suggestion which looks right